data/archipelago/scripts/item_utils.lua

```diff
@@ -49,25 +60,28 @@ function SpawnItem(item_id, traps)
 end
 
 
-function NGSpawnItems(item_list)
+function NGSpawnItems(item_counts)
 	local itemx = 595
 	local itemy = -90
 	local wandx = 600
 	local wandy = -120
-	local item_count = 0
-	for _, v in pairs(item_list) do
-		item_count = item_count + v
-	end
+	--local item_count = 0
+	--for _, v in pairs(item_counts) do
+	--	item_count = item_count + v
+	--end
 	-- check how many hearts are on the list, increase your health based on them, then remove them from the list
-	if item_list[AP.HEART_ITEM_ID] ~= nil then
-		local cur_hp, max_hp = get_health()
-		cur_hp = cur_hp + item_list[AP.HEART_ITEM_ID]
-		max_hp = max_hp + item_list[AP.HEART_ITEM_ID]
-		set_health(cur_hp, max_hp)
-		item_list[AP.HEART_ITEM_ID] = nil
+	if item_counts[AP.HEART_ITEM_ID] ~= nil or item_counts[AP.ORB_ITEM_ID] ~= nil then
+		local heart_amt = item_counts[AP.HEART_ITEM_ID] or 0
+		local orb_amt = item_counts[AP.ORB_ITEM_ID] or 0
+		if orb_amt > 0 then
+			GivePlayerOrbsOnSpawn(orb_amt)
+		end

```

Comment:
```suggestion
		GivePlayerOrbsOnSpawn(orb_amt)
```

Can move the condition to the function instead, just because this one is getting quite large.

Reply:
Even better, it was already there to begin with